src/services/auth/plugins/password/index.ts

```diff
@@ -58,6 +89,83 @@ const plugin: FastifyPluginAsync = async (fastify) => {
       });
     },
   );
+
+  /**
+   * Create a reset password request.
+   * This will send an email to the member in his langage with a link to reset the password.
+   * The link target a frontend route endpoint.
+   * The link will be valid for 24h.
+   * If the member does not exist, or does not have a password, the request will return success, to avoid leaking information.
+   * If the captcha is invalid the request will fail.
+   * @param email - Email of the member requesting the password reset link.
+   * @param captcha - Recaptcha response token.
+   * @returns 204 No Content if the request was successful.
+   */
+  fastify.post<{ Body: { email: string; captcha: string } }>(
+    '/reset-password-request',
+    { schema: postResetPasswordRequest },
+    async (request, reply) => {
+      const { email, captcha } = request.body;
+
+      await fastify.validateCaptcha(request, captcha, RecaptchaAction.ResetPassword, {
+        shouldFail: false,
+      });
+
+      // We can already return to avoid leaking timing information.
+      reply.status(StatusCodes.NO_CONTENT);
+      reply.send();
+
+      const resetPasswordRequest = await memberPasswordService.createResetPasswordRequest(
+        buildRepositories(),
+        email,
+      );
+      if (resetPasswordRequest) {
+        const { token, member } = resetPasswordRequest;
+        memberPasswordService.mailResetPasswordRequest(email, token, member.lang);
+        const action = {
+          member,
+          type: ActionTriggers.AskResetPassword,
+          view: Context.Auth,
+          extra: {},
+        };
+        // Do not await the action to be saved. It is not critical.
+        fastify.actions.service.postMany(member, buildRepositories(), request, [action]);
+      }
+    },
+  );
+
+  /**
+   * Solve the reset password request.
+   * This will force the password to be updated.
+   * A special token is required to perform this action. This token is sent by email to the member after creating a reset password request.
+   * If the password is not strong enough, the request will fail with an error 400 Bad Request.
+   * @param password - New password.
+   * @returns 204 No Content if the request was successful.
+   */
+  fastify.patch<{ Body: { password: string } }>(
+    '/reset-password-request',
+    {
+      schema: patchResetPasswordRequest,
+      preValidation: fastifyPassport.authenticate(PASSPORT_STATEGY_ID, { session: false }), // Session is not required.
+    },
+    async (request, reply) => {
+      const token: string = ExtractJwt.fromAuthHeaderAsBearerToken()(request)!;
```

Comment:
It looks like you do this already in the callback of the passport strategy, is there a reason why you do it again here ? is it not safe/possible to pass information needed down to the request handler function from the passport strategy callback ?